### PR TITLE
testing object instances

### PR DESCRIPTION
**Pull request by dennivision**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -148,29 +183,61 @@ struct Museum
...
+        {
+            std::cout << "a Museum::Visitor named '" << name << "' was created" << std::endl; 
+        }
+
+        float payAdmissionFee(float feeAmount)
+        {
+            ++timesVisited; // bit confused why this doesn't appear to update the timesVisited int
+            std::cout   << "Museum::Vistor:payAdmissionFee() - A visitor named '" << name << "' payed an admmission fee of $"
+                        << std::fixed << std::setprecision(2) << feeAmount << " and has now visited " << timesVisited << " times" << std::endl;
+            return feeAmount;
```
Comment (matkatmusic) [inline main.cpp]: call `std::setprecision(2)` at the top of `main()` so all `std::cout` code that follows will use this precision setting.
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -148,29 +183,61 @@ struct Museum
...
+Museum::Museum()
+{
+    std::cout << "a Museum object was created" << std::endl; 
+}
+
 float Museum::chargeVisitor(float amountToCharge, Museum::Visitor visitor)
 {
+    std::cout << "Museum::chargeVisitor() a vistor named '" << visitor.name << "' was charged $" << amountToCharge << std::endl; 
     return visitor.payAdmissionFee(amountToCharge);
 }
```
Comment (matkatmusic) [inline main.cpp:218]: This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course. If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.

File: main.cpp [hunk 3]
```diff
@@ -203,13 +277,17 @@ void SubwooferFactory::purchasePlywood(int sheets, float thickness)
     {
         plywoodInStock += (sheets * 8.f);
     }
+    {
+        std::cout << "SubwooferFactory::purchasePlywood() 3/4\" plywood is needed! " << thickness << " is not a valid thickness." << std::endl;
+    }
```
Comment (matkatmusic) [inline main.cpp]: This expression does not need to be nested inside curly braces.
remove the curly braces and fix the indentation
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (dennivision) [reply main.cpp:218]: got it, thanks

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**